core/server/master/src/main/java/alluxio/master/file/FileSystemMaster.java

```diff
@@ -829,8 +825,7 @@ public PersistenceState getPersistenceState(long fileId) throws FileDoesNotExist
       throws AccessControlException, FileDoesNotExistException, InvalidPathException {
     Metrics.GET_FILE_INFO_OPS.inc();
     try (JournalContext journalContext = createJournalContext();
-        LockedInodePath inodePath = mInodeTree.lockInodePath(path, InodeTree.LockMode.WRITE)) {
-      // This is WRITE locked, since loading metadata is possible.
+        LockedInodePath inodePath = mInodeTree.lockInodePath(path, InodeTree.LockMode.READ)) {
```

Comment:
Like the earlier response. `listStatus ` is a read operation, and the reason we used the WRITE lock before was to allow loading metadata. Now, the new locking scheme will handle creating inodes.